template/content/clients.js

```diff
@@ -0,0 +1,141 @@
+        return new ApplicationClients(config, options);
+    }
+
+    var self = this;
+
+    self.logger = options.logger || Logtron({
+        meta: {
+            team: 'my-team',
+            project: config.get('project')
+        },
+        backends: Logtron.defaultBackends({
+            console: config.get('clients.logtron.console'),
+            raw: true
+        })
+    });
+    self.statsd = options.statsd || new Statsd({
+        host: config.get('clients.uber-statsd-client.host'),
+        port: config.get('clients.uber-statsd-client.port'),
+        prefix: [
+            config.get('project'),
+            process.env.NODE_ENV,
+            os.hostname().split('.')[0]
+        ]
+    });
+    self.uncaught = Uncaught({
+        logger: self.logger,
+        statsd: self.statsd,
+        statsdKey: 'uncaught-exception',
+        prefix: [
+            config.get('project'),
+            process.env.NODE_ENV,
+            os.hostname().split('.')[0]
+        ],
+        backupFile: config.get('clients.uncaught-exception.backupFile')
+    });
+    self.bootFile = config.get('bootFile');
+    self.port = config.get('port');
+    self.host = config.get('host') || myLocalIp();
+
+    self.rootChannel = TChannel({
+        logger: self.logger,
+        statsd: self.statsd,
+        statTags: {
+            app: SERVICE_NAME
+        }
+    });
+
+    self.appChannel = self.rootChannel.makeSubChannel({
+        serviceName: SERVICE_NAME
+    });
+    self.ringpopChannel = self.rootChannel.makeSubChannel({
+        serviceName: 'ringpop'
+    });
+    self.tchannelThrift = self.rootChannel.TChannelAsThrift({
+        source: thriftFile
+    });
+
+    self.hyperbahnClient = HyperbahnClient({
+        tchannel: self.rootChannel,
+        serviceName: SERVICE_NAME,
+        hostPortList: config.get('clients.hyperbahn.seedList'),
+        hardFail: config.get('clients.hyperbahn.hardFail'),
+        reportTracing: config.get('clients.hyperbahn.reportTracing'),
+        logger: self.logger,
+        statsd: self.statsd
+    });
+
+    self.ringpop = Ringpop({
+        app: SERVICE_NAME,
+        logger: self.logger,
+        channel: self.ringpopChannel,
+        hostPort: self.host + ':' + self.port
+    });
+}
+
+ApplicationClients.prototype.bootstrap = function bootstrap(cb) {
+    var self = this;
+
+    self.rootChannel.listen(self.port, self.host, onListening);
+
+    function onListening() {
+        self.ringpop.setupChannel();
+
+        if (self.bootFile) {
+            self.ringpop.bootstrap(self.bootFile, onRingpop);
+        } else {
+            onRingpop(null);
+        }
+    }
+
+    function onRingpop(err) {
+        /*istanbul ignore if*/
+        if (err) {
+            self.logger.fatal('cannot bootstrap ringpop', {
+                err: err
+            });
+            return cb(err);
+        }
+
+        self.hyperbahnClient.once('advertised', cb);
+        self.hyperbahnClient.advertise();
```

Comment:
Do you catch the error from this anywhere? I think the process is not going to quit if there's an error advertising